io-webp.c

```diff
@@ -12,10 +12,28 @@
 
 #include "io-webp.h"
 
-#if defined(HAVE_ENDIAN_H)
+/* endian.h is non-standard and platforms disagree with each other on it.
+ *
+ * * GLIBC (and its family) supports <endian.h>.
+ *
+ * * BSD libc provides <sys/endian.h> instead (Note that OpenBSD since
+ *   5.6 supports conventional function names).
+ *
+ * * macOS does not support <endian.h> but exposes another set of macros
+ *   in libkern.
+ */
+#if defined(__GLIBC__)
 #include <endian.h>
-#elif defined(HAVE_SYS_ENDIAN_H)
+#elif defined(__OpenBSD__) || defined(__FreeBSD__) || defined(__NetBSD__)
 #include <sys/endian.h>
+#elif defined(__APPLE__)
+#include <libkern/OSByteOrder.h>
+#define le16toh(x) OSSwapLittleToHostInt16(x)
+#define le32toh(x) OSSwapLittleToHostInt32(x)
```

Comment:
le32toh is the only function needed, I'd only define that one, the rest is uneeded boilerplate that can be confusing

Reply:
Sounds good to me, I was thinking about the same.